nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -424,6 +423,40 @@ public static void checkTypeParameterNullnessForAssignability(
     if (tree instanceof VariableTree) {
       VariableTree varTree = (VariableTree) tree;
       rhsTree = varTree.getInitializer();
+      Tree lhsTypeTree = varTree.getType();
+
+      if (rhsTree instanceof MethodInvocationTree) {
+        MethodInvocationTree methodInvocationTree = (MethodInvocationTree) rhsTree;
+        Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(methodInvocationTree);
+        // update inferredTypes cache for assignments
+        if (methodSymbol.type instanceof Type.ForAll // generic method call
+            && methodInvocationTree.getTypeArguments().isEmpty() // no explicit generic arguments
+            && lhsTypeTree instanceof ParameterizedTypeTree) { // lhs type has generic
```

Comment:
Rather than checking the tree, can we check if `lhsType` is generic?  That might be more general.